File: test/bench/b4.lua
-- Lua transcription of b4 (variable r/m/w, 50 M iterations).
-- Run: lua5.4 b4.lua
-- Uses a 1-element table to force table-load and table-store per
-- iteration, matching the Forth `v @ 1 + v !` heap r/m/w pattern.
local v = {0}
for i = 1, 50000000 do v[1] = v[1] + 1 end

File: test/bench/b4.py
# Python 3 transcription of b4 (variable r/m/w, 50 M iterations).
# Run: python3 b4.py
# Uses a 1-element list to force list-load and list-store per
# iteration, matching the Forth `v @ 1 + v !` heap r/m/w pattern and
# the Lua 1-element-table transcription.
v = [0]
for i in range(50000000):
    v[0] = v[0] + 1
